cassovary-core/src/main/scala/com/twitter/cassovary/graph/SortedNeighborsNodeOps.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2014 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.graph
+
+import com.twitter.cassovary.util.SortedArrayOps
+
+trait SortedNeighborsNodeOps {
+  self: Node =>
+
+  override protected def containsNode(nodeIds: Seq[Int], queryNodeId: Int): Boolean = {
```

Comment:
It is clear from these examples that the choice of method is based on
compile-time type... No good for us.

On Sun, Feb 1, 2015 at 12:08 PM, Szymon <notifications@github.com> wrote:

> In
> cassovary-core/src/main/scala/com/twitter/cassovary/graph/SortedNeighborsNodeOps.scala
> <https://github.com/twitter/cassovary/pull/133#discussion_r23900672>:
>
> > + *
> > + *     http://www.apache.org/licenses/LICENSE-2.0
> > + *
> > + * Unless required by applicable law or agreed to in writing, software distributed
> > + * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
> > + * CONDITIONS OF ANY KIND, either express or implied. See the License for the
> > + * specific language governing permissions and limitations under the License.
> > + */
> > +package com.twitter.cassovary.graph
> > +
> > +import com.twitter.cassovary.util.SortedArrayOps
> > +
> > +trait SortedNeighborsNodeOps {
> > +  self: Node =>
> > +
> > +  override protected def containsNode(nodeIds: Seq[Int], queryNodeId: Int): Boolean = {
>
> I didn't know about that.
>
> But we still have a problem here. I used as before but:
>
> object A3 extends A2 {
>   val arr: Seq[Int] = Array(0)
>   println("In Seq:" + f(Seq(1,2))); println("In Array:" + f(arr))
> }
>
> and obtained:
>
> // Exiting paste mode, now interpreting.
>
> defined object A3
>
> scala> A3
> false
> In Seq:true
> In Array:true
> res0: A3.type = A3$@51b69452
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twitter/cassovary/pull/133/files#r23900672>.
>